mne/beamformer/_lcmv.py

```diff
@@ -338,6 +338,100 @@ def apply_lcmv_epochs(epochs, filters, *, return_generator=False,
     return stcs
 
 
+@verbose
+def apply_lcmv_evoked_tfr(evoked_tfr, filters, verbose=None):
+    """Apply Linearly Constrained Minimum Variance (LCMV) beamformer weights.
+
+    Apply Linearly Constrained Minimum Variance (LCMV) beamformer weights
+    on single trial time-frequency data.
+
+    Parameters
+    ----------
+    evoked_tfr : AverageTFR
+        Time-frequency resolved evoked.
+    filters : list of Beamformer
+        LCMV spatial filter (beamformer weights)
+        Filter weights returned from :func:`make_lcmv` for each frequency.
+    %(verbose)s
+
+    Returns
+    -------
+    stc : TFRSourceEstimate
+        The source estimate.
+
+    See Also
+    --------
+    make_lcmv, apply_lcmv_epochs, apply_lcmv_raw, apply_lcmv, apply_lcmv_cov
+    """
+    info = evoked_tfr.info
+    tmin = evoked_tfr.times[0]
+    these_filters = filters[0]
+    n_orient = 3 if these_filters['pick_ori'] == 'vector' else 1
+    data = np.empty(shape=(evoked_tfr.freqs.size, 1,
+                           these_filters['n_sources'], n_orient,
+                           evoked_tfr.times.size))
+    for freq_idx, these_filters in enumerate(filters):
+        sel = _check_channels_spatial_filter(
+            evoked_tfr.ch_names, these_filters)
+        evoked_data = evoked_tfr.data[sel, freq_idx, :]
+        stc = next(_apply_lcmv(data=evoked_data, filters=these_filters,
+                               info=info, tmin=tmin))
+        data[freq_idx, 0] = stc.data[:, None] if n_orient == 1 else stc.data
+
+    # swap frequency to second to last dimension
+    return TFRSourceEstimate(data=data.transpose([1, 2, 3, 0, 4]), info=info,
+                             vertices=these_filters['vertices'],
+                             times=evoked_tfr.times, freqs=evoked_tfr.freqs)
+
+
+@verbose
+def apply_lcmv_epochs_tfr(epochs_tfr, filters, verbose=None):
```

Comment:
Same as above: wouldn't call it that, and why do we need an extra function?

Reply:
Because you are passing the epochs that have a bandfiltered frequency so the filters need to be looped over, one for each frequency. It hacks the MNE structure and uses Epochs to feed into the `_apply_lcmv` as we do in other places in the codebase but that kind of stuff really can't be in examples.